src/plugins/comments/test/comments.e2e.js

```diff
@@ -546,5 +615,50 @@ describe('Comments', () => {
 
       expect(afterSetCellMetaCallback).toHaveBeenCalledWith(0, 0, 'comment', { value: 'Edited comment' }, undefined, undefined);
     });
+
+    it('should not trigger `afterSetCellMeta` callback after editing comment by context menu if `beforeSetCellMeta` returned false', async() => {
+      const afterSetCellMetaCallback = jasmine.createSpy('afterSetCellMetaCallback');
+
+      handsontable({
+        data: Handsontable.helper.createSpreadsheetData(10, 10),
+        rowHeaders: true,
+        colHeaders: true,
+        contextMenu: true,
+        comments: true,
+        columns() {
+          return {
+            comment: {
+              value: 'test'
+            }
+          };
+        },
+        beforeSetCellMeta: () => false,
+        afterSetCellMeta: afterSetCellMetaCallback
+      });
+
+      selectCell(0, 0);
+      contextMenu();
+
+      const editCommentButton = $('.htItemWrapper').filter(function() {
+        return $(this).text() === 'Edit comment';
+      })[0];
+
+      $(editCommentButton).simulate('mousedown');
+      $(editCommentButton).simulate('mouseup');
+
+      const textarea = spec().$container[0].parentNode.querySelector('.htCommentTextArea');
+      textarea.focus();
+      textarea.value = 'Edited comment';
+
+      await sleep(100);
+
+      $('body').simulate('mousedown');
+      $('body').simulate('mouseup');
+      textarea.blur();
+
+      await sleep(400);
+
+      expect(afterSetCellMetaCallback).not.toHaveBeenCalled();
```

Comment:
I'd remove `expect(afterSetCellMetaCallback).not.toHaveBeenCalled();` As in my previous comments it's not a good place to tests implementation details of the `setCellMeta`. Replacing that line to something like `expect(getCellMeta(0, 0).comments).toBe....` is more specyfic to the plugin.